packages/valory/skills/proposal_collector_abci/rounds.py

```diff
@@ -135,10 +142,18 @@ def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Event]]:
                 if not existing:
                     delegations.append(nd)
 
+                # Do we need to refresh any proposal?
+                for p in proposals.values():
+                    if nd["token_address"] in p["governor"]["tokens"][0]["id"]:
+                        proposals_to_refresh.add(p["id"])
```

Comment:
It looks like this logic should belong to some behaviour.

Reply:
I think we should leave it here: we need access to the different new delegations from every agent (this round is CollectDifferent), which  we aggregate here.

Reply:
I see, it makes sense to keep this here as we avoid an extra round.